Resources/Computing/HTCondorCEComputingElement.py

```diff
@@ -516,6 +522,16 @@ def __cleanup(self):
     # FIXME: again some issue with the working directory...
     # workingDirectory = self.ceParameters.get( 'WorkingDirectory', DEFAULT_WORKINGDIRECTORY )
 
+    if not self._cleanupLock.acquire(False):
+      return
+
+    now = datetime.datetime.utcnow()
+    if (self._lastCleanupTime - now).total_seconds < 60:
```

Comment:
Is it really needed this often? If we're keeping 2 hours worth of files I suspect it can be done a few times an hour at most.

Reply:
I suspect it can also be done once every minute, but anyway this is limited by the cycle factor of pilot status update as well.
How long does a single find run in your folder?

Reply:
We've had concurrent `find` running for days.

Reply:
But then it should be locked out as long at it is running. And it should be faster if there are less files.

Reply:
> We've had concurrent find running for days.
Do you mean a single find was running for days, or that the find's were running concurrently for days?
Because first one should have blocked the SiteDirector? Or is there an implicit timeout in the Execution of updatePilotStatus?